### PR TITLE
adding options for windows setup.py

### PR DESCRIPTION
**Pull request by fcollman**
I got this to compile on windows today with these options.. one could use appveyor maybe to add this to testing.  I had to install visual studio community edition 2017 with c++ options, and then install scikit-build and then it worked.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -30,9 +30,22 @@
...
+    extra_link_args = ['/LIBPATH:{0}'.format(lib_dir)]+ library_link_args
+    extra_compile_args = [
+              '/std:c++17','-O3'
+            ]
+else:
+    extra_link_args = ['-L{0}'.format(lib_dir)] 
+    extra_compile_args = [
+              '-std=c++11','-O3'
+            ]
+if sys.platform == 'win32':
```
Comment (manuel-castro) [inline setup.py]: Line 48 is a typo right?

**Comment by william-silversmith** [comment]
I tried compiling this on Python3.7 in a Windows 10 VM and I got an error:

Cannot open include file: "draco/compression/decode.h": No such file or directory in DracoPy.h

**Comment by fcollman** [comment]
so i couldn't get the pip install to work either.. instead I had to check it out from source, and git submodule init and git submodule update to check out the draco code too, the compile with setup.py

File: setup.py [hunk 2]
```diff
@@ -30,9 +30,22 @@
...
+    library_link_args = ['{0}'.format(lib) for lib in ('dracoenc.lib', 'draco.lib', 'dracodec.lib')]
 else:
-    library_link_args = ['-l:{0}'.format(lib) for lib in ('libdracoenc.a', 'libdraco.a', 'libdracodec.a')]
-extra_link_args = ['-L{0}'.format(lib_dir)] + library_link_args
+    library_link_args = ['-l{0}'.format(lib) for lib in ('libdracoenc.a', 'libdraco.a', 'libdracodec.a')]
+
+if sys.platform == 'win32':
+    extra_link_args = ['/LIBPATH:{0}'.format(lib_dir)]+ library_link_args
+    extra_compile_args = [
+              '/std:c++17','-O3'
```
[the remark this answers is not in the document]
Comment (fcollman) [reply setup.py]: not sure.. i just read on the internet that the c++17 was the visual studio version of the c++11 flag and tried it.

File: setup.py (hunk 1 above)
Comment (fcollman) [reply setup.py]: indeed.. fixed.

Comment (fcollman) [reply setup.py]: in truth i got this working on a windows machine directly editing the file to make it work for windows and then on my laptop tried to make it more general, but i haven't been back to a windows machine to validate it.

**Comment by manuel-castro** [comment]
version 0.0.15 has this update

**Comment by jefferis** [comment]
On windows with python looking like this:

```
> py_config()
python:         C:/Users/jefferis/AppData/Local/r-miniconda/envs/r-reticulate/python.exe
libpython:      C:/Users/jefferis/AppData/Local/r-miniconda/envs/r-reticulate/python36.dll
pythonhome:     C:/Users/jefferis/AppData/Local/r-miniconda/envs/r-reticulate
version:        3.6.10 |Anaconda, Inc.| (default, May  7 2020, 19:46:08) [MSC v.1916 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:/Users/jefferis/AppData/Local/r-miniconda/envs/r-reticulate/Lib/site-packages/numpy
numpy_version:  1.18.
```

and getting this with pip install

```
  ----------------------------------------
  ERROR: Failed building wheel for DracoPy
Failed to build DracoPy
ERROR: Could not build wheels for DracoPy which use PEP 517 and cannot be installed directly
```

Am I supposed to be able to pip install on Windows at this point? Or do i need to check out the repo etc as suggested by @fcollman on 16 Oct 2019?

**Comment by manuel-castro** [comment]
Could you please try `pip install DracoPy --no-binary DracoPy`

**Comment by anhducle4** [comment]
With Python on my Windows looking like this

```
[python:         C:/Users/anhdu/AppData/Local/Programs/Python/Python38/python.exe
libpython:      C:/Users/anhdu/AppData/Local/Programs/Python/Python38/python38.dll
pythonhome:     C:/Users/anhdu/AppData/Local/Programs/Python/Python38
version:        3.8.4 (tags/v3.8.4:dfa645a, Jul 13 2020, 16:46:45) [MSC v.1924 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:/Users/anhdu/AppData/Local/Programs/Python/Python38/Lib/site-packages/numpy
numpy_version:  1.19.0]

```

and tried `pip install DracoPy --no-binary DracoPy` using Visual Studio Build Tools 2019 following this [](https://www.scivision.dev/python-windows-visual-c-14-required/ ) **I still get the same end message as Dr. Jefferis mentioned above.**

**Comment by manuel-castro** [comment]
I spent a decent amount of time banging my head against the wall trying to solve this to no avail. It seems like there are some issues with the scikit-build package and Visual Studio. See: https://github.com/scikit-build/scikit-build/issues/455 and https://github.com/scikit-build/scikit-build/issues/496. There is likely a way to get it to work, but I recommend using Windows Subsystem for Linux instead. If you must use Windows proper, the right path might be to try creating an environment that mirrors Forrest's previously working solution (Visual Studio 2017 and scikit-build==0.0.10), but I was not able to do that successfully.

**Comment by jefferis** [comment]
Thanks so much for looking into this @manuel-castro.

@anhducle4 I am afraid I have consistently had more installation dependency troubles on Windows for cloudvolume, DracoPy and meshparty. The new WSL2 may be a good option with Windows >=2004 but I have not investigated at all. Currently I think it's not completely straightforward to run a GUI application (such as rstudio or a python IDE) running on top of WSL1 although I have seen a few [howtos](https://techcommunity.microsoft.com/t5/windows-dev-appconsult/running-wsl-gui-apps-on-windows-10/ba-p/1493242) when [googling](https://www.google.com/search?q=wsl+gui&oq=wsl). Therefore for the specific task of fetching flywire meshes on Windows, you may find it saves time to fetch and save the meshes as obj files using a minimal installation under WSL1. You can then use a GUI app running under Windows to read in and analyse the obj mesh files.

**Comment by manuel-castro** [comment]
I will leave a note in the README that this is broken, and try again to fix it later at some point. Perhaps a generous magician will bless us with a working solution at some point, like Forrest did last time.

**Comment by anhducle4** [comment]
Thank you @manuel-castro for taking a look at this and thank you @jefferis for your guidance!

**Comment by manuel-castro** [comment]
Thanks to @william-silversmith we now have working Windows binaries on PyPI once again!

**Pull request merged**